chimney/src/main/scala/io/scalaland/chimney/dsl/TransformerDefinition.scala

```diff
@@ -7,37 +7,122 @@ import io.scalaland.chimney.internal.macros.{ChimneyBlackboxMacros, TransformerD
 
 import scala.language.experimental.macros
 
+/** Allows customization of [[io.scalaland.chimney.Transformer]] derivation
+  *
+  * @tparam From data type that will be used as input
+  * @tparam To   data type that will be used as output
+  * @tparam C    type-level encoded config
+  */
 final class TransformerDefinition[From, To, C <: TransformerCfg](
     val overrides: Map[String, Any],
     val instances: Map[(String, String), Any]
 ) {
+
+  /** Fail derivation if `From` type is missing field even if `To` has default value for it
+    *
+    * By default in such case derivation will fallback to default values.
+    *
+    * @see [[https://scalalandio.github.io/chimney/#Usage]] for more details
+    * @return [[io.scalaland.chimney.dsl.TransformerDefinition]]
+    */
   def disableDefaultValues: TransformerDefinition[From, To, DisableDefaultValues[C]] =
     this.asInstanceOf[TransformerDefinition[From, To, DisableDefaultValues[C]]]
 
+  /** Enable Java Beans naming convention (`.getName`, `.isName`) on `From`
+    *
+    * By default only Scala conversions (`.name`) are allowed.
+    *
+    * @see [[https://scalalandio.github.io/chimney/#Usage]] for more details
+    * @return [[io.scalaland.chimney.dsl.TransformerDefinition]]
+    */
   def enableBeanGetters: TransformerDefinition[From, To, EnableBeanGetters[C]] =
     this.asInstanceOf[TransformerDefinition[From, To, EnableBeanGetters[C]]]
 
+  /** Enable Java Beans naming convention (`.setName(value)`) on `To`
+    *
+    * By default only Scala conversions (`.copy(name = value)`) are allowed.
+    *
+    * @see [[https://scalalandio.github.io/chimney/#Usage]] for more details
+    * @return [[io.scalaland.chimney.dsl.TransformerDefinition]]
+    */
   def enableBeanSetters: TransformerDefinition[From, To, EnableBeanSetters[C]] =
     this.asInstanceOf[TransformerDefinition[From, To, EnableBeanSetters[C]]]
 
+  /** Allow usage of None if field is missing in `From`, no rename/const value/calulated
```

Comment:
This explanation is a bit inaccurate. The flag does not "allow usage of None", because it's always can be used and we don't need to enable it anyhow. 

More precise description would be: `Sets target value of optional field to None if field is missing from source type From` (or sth similar).